### PR TITLE
Binding model: Bindless part 1

### PR DESCRIPTION
**Pull request by doitsujin**
As requested.

### CODE REVIEW

**Comment by doitsujin** [comment]
Could add the descriptor offsets to 21a1bad as discussed in #20 .

**Comment by HansKristian-Work** [comment]
> Could add the descriptor offsets to [21a1bad](https://github.com/HansKristian-Work/vkd3d/commit/21a1badfacf4be4b0895da5233ff692868ffa281) as discussed in #20 .

yes, please.

**Comment by doitsujin** [comment]
Done.

File: libs/vkd3d/vkd3d_private.h [hunk 1]
```diff
@@ -618,8 +625,38 @@ struct d3d12_descriptor_heap
...
+
+static inline unsigned int d3d12_descriptor_heap_uav_set_index(bool is_buffer)
+{
+    return 3 + (is_buffer ? 0 : 1);
+}
+
+static inline uint32_t d3d12_desc_heap_offset(const struct d3d12_desc *dst)
+{
+    size_t heap_start = (size_t)dst->heap->descriptors;
+    return ((size_t)dst - heap_start) / sizeof(struct d3d12_desc);
```
Comment (HansKristian-Work) [inline libs/vkd3d/vkd3d_private.h]: This is still here.

Comment (doitsujin) [reply libs/vkd3d/vkd3d_private.h]: Which commit is this? I thought this was later.

Comment (HansKristian-Work) [reply libs/vkd3d/vkd3d_private.h]: 7c20b0a3c4fed016444d8e433fa470000be0fd88

Comment (doitsujin) [reply libs/vkd3d/vkd3d_private.h]: Ok, found it, fixed.

**Pull request merged**